tests/utils.go

```diff
@@ -100,3 +123,27 @@ func GetKubeClientFromRESTConfig(config *rest.Config) (*kubernetes.Clientset, er
 
 	return coreClient, nil
 }
+
+func GenerateNamespace() string {
```

Comment:
I like this approach but we need to reconcile it with recently merged pr #314 https://github.com/kubevirt/containerized-data-importer/blob/master/tests/utils.go#L22 where the test hard-codes the ns

Reply:
Its fine to use hard coded namespaces.  We'll need more than one though in order to test cloning across multiple namespaces.

Reply:
That is the install namespace, where when we don't specify a different name space in the manifest generation the cdi stuff is installed. It is setup as an argument to the test runner so we can override it without recompiling the code.

Reply:
@davidvossel 
> Its fine to use hard coded namespaces. We'll need more than one though in order to test cloning across multiple namespaces.

Is this still true when using ginkgo in parallel?

Reply:
@davidvossel wrote:
> ts fine to use hard coded namespaces.

How will that scale and work when tests are run in parallel?

Reply:
we'd have to randomize if we start running in parallel.

I'm not sure we'll want to run in parallel though. KubeVirt doesn't.

If we test in parallel, it impacts the timeouts associated with our tests. It becomes difficult to distinguish between an actual performance degrading timing issue and one that is just caused by multiple intensive tests running at the same time.  Also, parallel is more difficult to debug. It's the difference from having a controller log that's isolated to a single test and one that's got lots of unrelated things going on within it.

These functional tests are difficult to keep stable at times. Each k8s and os provider behaves slightly differently. Anything we can do to keep the execution of these tests as simple as possible will work in our favor.